MatrixSDK/MXSession.m

```diff
@@ -1598,6 +1638,32 @@ - (void)handleToDeviceEvent:(MXEvent *)event
     return roomsInSyncResponse;
 }
 
+- (void)handleSyncResponseIfRequiredWithCompletion:(void (^)(void))completion
+{
+    NSLog(@"[MXSession] handleSyncResponseIfRequired. state %tu", _state);
+    
+    MXSyncResponseFileStore *syncResponseStore = [[MXSyncResponseFileStore alloc] init];
+    [syncResponseStore openWithCredentials:self.credentials];
+    if (syncResponseStore.syncResponse)
+    {
+        [self handleSyncResponse:syncResponseStore.syncResponse
```

Comment:
We should add more check before trusting MXSyncResponseFileStore data.

I am a bit scared with services running in parallel and race conditions. We must have continuous data in MXSession. This continuity can be broken if MXBackgroundSyncService and MXSession got desync'ed.

One way to avoid this is to make sure that the data from  MXSyncResponseFileStore corresponds to the token we expect (`MXStore.eventStreamToken`). This means that MXSyncResponseFileStore should store the token from which it started to gather data.

If tokens do not match, we trash MXSyncResponseFileStore data.